src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,112 @@
+package duke;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.TaskType;
+import duke.task.ToDo;
+
+/**
+ * Deals with loading tasks from the file and saving tasks in the file.
+ */
+public class Storage {
+    private final String filePath;
+
+    /**
+     * Constructs the storage.
+     *
+     * @param filePath the filepath of the storage file
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Reads the data of tasks from previous session's file as TaskList.
+     * If there's no file yet, create an empty file, and return empty list.
+     *
+     * @return the list of tasks from previous session
+     */
+    public TaskList readFile() {
+        List<Task> taskList = new ArrayList<>();
+        try {
+            File file = new File(filePath);
+            if (!file.createNewFile()) {
+                File myObj = new File(filePath);
+                Scanner myReader = new Scanner(myObj);
```

Comment:
Do you think that extracting the codes for converting the data file to its tasks could be clearer for the readers?